### ISSUE TITLE
Dragonstrike

### ISSUE THREAD

**Issue by thenumberpurplee** [SunwellWoW/Sunwell-TBC-Bugtracker#1932]
Decription: 
When using 2 Dragonstrike you can stack the buffs

How it works: 

When wearing 2 Dragonstrikes if they both proc at the same time you get 2 haste buffs
How it should work: 

If the second Dragonstrike procs while you have the buff from the first Dragonstrike it should refresh the buff instead of stacking and giving you two
Source (you should point out proofs of your report, please give us some source):

**Comment by sicknot1** [comment]
are you sure 100%? 
https://www.wowhead.com/item=28439/dragonstrike#comments:id=287394
This is all i can found and it's 2.4.2 live. Meaning they should stack before that patch. Or not.

**Comment by thenumberpurplee** [comment]
i looked at my haste rating and they were stacking. Did not just see 2 bufss

**Comment by alexnxs** [comment]
"They still have no CD, even single wielding one gets you refreshing buffs quite often, but I had not a single stacking buff from MH + OH. I never got below 1.9 hasted speed."

http://elitistjerks.com/f81/t12386-future_dps_warrior/p143/#post737042

Prior to 2.4.2  these weapons were main hand only weapon and not dual wieldable. 
They stacked on early 2.4.2 PTR but it became apparent how ridiculous this would be, plus no other weapon buffs stack with themselves.

Some other comments in that thread too where people tested:
http://elitistjerks.com/f81/t12386-future_dps_warrior/p145/#post744170
"Well I just tested dw dragonstrike/dragonmaw on live and the haste effect didn't stack, just renewed it. Spent 15 minutes on a servant in BL, The quickest my attack speed went down to was 1.9 from 2.7 (flurry + haste proc)"

Proc rate seems to be good, judging from comments in same thread, someone tried to model it:

"a ppm of 1.4 and that each proc will give 4 hasted swings (in reality it's 4.2 swings) I get an uptime of 35% for dual DS with overwriting buffs. I calculated it using a quick and dirty python script over 1000 swings and only using white swings. Specials will complicate things quite a lot in my script so I don't have time to add them, but they will of course make the uptime go up a bit, maybe up towards 40+%, but I don't think it's very likely that you will reach 60% uptime.
